packages/ia-components/sandbox/details/AnchorSearch.jsx

```diff
@@ -63,21 +63,24 @@ class AnchorSearch extends IAReactComponent {
 
   constructor(props) {
     super(props); // { query || field & value, sort, reload }
-    //TODO-STATE this might have the issue of constructor not being re-run and needing componentDidMount catch
-    this.setState({
-      query: this.props.query || ObjectFromEntries([[this.props.field, this.props.value]]),
-      urlProps: ObjectFilter(this.props, (k, unusedV) => AnchorSearch.urlparms.includes(k)),  //sort, reload, query
-      anchorProps: ObjectFilter(this.props, (k, unusedV) => (!AnchorSearch.urlparms.includes(k) && !['children'].includes(k)))
-    });
+    this.onClick = this.onClick.bind(this);
+    this.state = {};
```

Comment:
what's your constructor preference, setting state before or after the event bindings?

Reply:
Is there are reason for a preference ? It should be exactly equivalent.

Reply:
just pickin' your brain around code organization.